bgpd/bgp_mplsvpn.c

```diff
@@ -1616,14 +1618,40 @@ void vrf_import_from_vrf(struct bgp *to_bgp, struct bgp *from_bgp,
 	 */
 	vname = (from_bgp->name ? XSTRDUP(MTYPE_TMP, from_bgp->name)
 			       : XSTRDUP(MTYPE_TMP, VRF_DEFAULT_NAME));
+	if (!listcount(to_bgp->vpn_policy[afi].import_vrf))
+		listnode_add(to_bgp->vpn_policy[afi].import_vrf, vname);
+	else {
+		is_inst_match = false;
+		for (ALL_LIST_ELEMENTS_RO(to_bgp->vpn_policy[afi].import_vrf,
+					  node, tmp_name)) {
+			if (strcmp(vname, tmp_name) == 0) {
+				is_inst_match = true;
+				break;
+			}
+		}
+		if (!is_inst_match)
+			listnode_add(to_bgp->vpn_policy[afi].import_vrf,
+					     vname);
+	}
 
-	listnode_add(to_bgp->vpn_policy[afi].import_vrf, vname);
-
-	if (!listcount(from_bgp->vpn_policy[afi].export_vrf))
-		first_export = true;
 	vname = XSTRDUP(MTYPE_TMP, export_name);
-	listnode_add(from_bgp->vpn_policy[afi].export_vrf, vname);
-
+	if (!listcount(from_bgp->vpn_policy[afi].export_vrf)) {
```

Comment:
This is unnecesarrily complicated code.  the else statement part of this can handle everything and should just be what we need here.